pkg/tunnelserver/tunnel.go

```diff
@@ -116,6 +116,9 @@ func (t *Authorizer) Authorize(req *http.Request) (*Client, bool, error) {
 
 	if input.Node != nil {
 		node, ok, err := t.authorizeNode(cluster, input.Node, req)
+		if err != nil {
```

Comment:
in here if err is not nil we return a nil client but the original code doesn't. I don't know if this makes a difference but we might need to check.

Reply:
if the error happen, the authorizeNode return nil client, if we continue, may panic below. So I add the check here.